c10/core/Scalar.h

```diff
@@ -30,12 +31,23 @@ namespace c10 {
  * but don't know its type.
  */
 class C10_API Scalar {
+ public:
  public:
   Scalar() : Scalar(int64_t(0)) {}
 
 #define DEFINE_IMPLICIT_CTOR(type, name) \
   Scalar(type vv) : Scalar(vv, true) {}
 
+  void destroy() {
+    if (Tag::HAS_si == tag) {
```

Comment:
I think we run into performance issues when we did something similar for `SymInt`. It might be less of an issue for Scalars. Tagging @swolchok

Reply:
This shouldn't be an issue for scalars as we do not create and destroy large numbers of them, unlike tensors.